Sources/App/Commands/Analyze.swift

```diff
@@ -143,6 +143,9 @@ func analyze(client: Client,
                          threadPool: threadPool,
                          transaction: tx,
                          packages: packages)
+                .map { throttleBranchVersions(
+                    packageDeltas: $0,
+                    delay: Constants.branchVersionRefreshDelay) }
```

Comment:
This is the core of it: after determining the `VersionDelta` (versions to be added, deleted, kept), we go through the three arrays and remove any recently changes `.branch` versions from `toAdd` and move the existing ones from `toDelete` to `toKeep`, essentially undoing any branch changes detected if they happened within the configured time window.